packages/story-editor/src/components/fontCheck/fontCheckDialog.js

```diff
@@ -0,0 +1,194 @@
+  const onChange = useCallback((font) => {
+    setSuggestedFont({ font });
+  }, []);
+
+  const onSettingsClick = useCallback((evt) => {
+    trackClick(evt, 'click_font_settings');
+  }, []);
+
+  const updateMissingFontWithDefault = useCallback(() => {
+    const { id, ...font } = defaultFont.font;
+    updateElementsByFontFamily({
+      family: missingFont,
+      properties: { font },
+    });
+    onClose();
+    // @todo handle ESC
+    // https://github.com/GoogleForCreators/web-stories-wp/pull/10808#discussion_r821817302
+  }, [onClose, missingFont, updateElementsByFontFamily]);
+
+  const updateMissingFontWithSelected = useCallback(() => {
+    const { id, ...font } = suggestedFont.font;
+    updateElementsByFontFamily({
+      family: missingFont,
+      properties: { font },
+    });
+    onClose();
+  }, [suggestedFont, onClose, missingFont, updateElementsByFontFamily]);
+
+  const Chevron = isRTL ? ChevronRight : ChevronLeft;
+
+  return (
+    <Dialog
+      isOpen={isOpen}
+      onClose={onClose}
+      title={__('Missing Fonts', 'web-stories')}
+      contentLabel={__('Missing Fonts', 'web-stories')}
+      actions={
+        <>
+          <Link href={dashboardLink}>
+            <DashboardLink>
+              <Chevron />
+              {__('Back to dashboard', 'web-stories')}
+            </DashboardLink>
+          </Link>
+          <Button
+            type={BUTTON_TYPES.SECONDARY}
+            size={BUTTON_SIZES.SMALL}
+            onClick={updateMissingFontWithDefault}
+          >
+            {__('Open anyway', 'web-stories')}
+          </Button>
+          <Button
+            type={BUTTON_TYPES.PRIMARY}
+            size={BUTTON_SIZES.SMALL}
+            onClick={updateMissingFontWithSelected}
+          >
+            {__('Replace font', 'web-stories')}
+          </Button>
+        </>
+      }
+    >
+      <DialogContent>
+        <Text size={THEME_CONSTANTS.TYPOGRAPHY.PRESET_SIZES.SMALL}>
+          {__(
+            'This story contains the following font that is no longer available:',
+            'web-stories'
+          )}
+        </Text>
+        <Text size={THEME_CONSTANTS.TYPOGRAPHY.PRESET_SIZES.SMALL} isBold>
+          {missingFont}
+        </Text>
+        <div>
+          <TranslateWithMarkup
+            mapping={{
+              a: (
+                <Link
+                  size={THEME_CONSTANTS.TYPOGRAPHY.PRESET_SIZES.SMALL}
+                  onClick={onSettingsClick}
+                  href={dashboardSettingsLink}
+                  isBold
+                />
+              ),
+            }}
+          >
+            {__(
+              'To replace this font with your own or re-add it, go to <a>Settings</a>. To replace this font with a default, open the story.',
+              'web-stories'
+            )}
+          </TranslateWithMarkup>
+        </div>
+        <Text size={THEME_CONSTANTS.TYPOGRAPHY.PRESET_SIZES.SMALL}>
+          {__('Or, choose replacement font below:', 'web-stories')}
```

Comment:
Does this make it hard to translate when strings are like this.

Reply:
This is fine because it's not really used for string concatenation.

In any case, the wording will most likely be tweaked in the future anyway.